engine/src/main/java/org/hibernate/validator/internal/constraintvalidators/bv/pattern/BasePatternValidator.java

```diff
@@ -40,12 +43,4 @@ public void initialize(Pattern parameters) {
 		}
 	}
 
-	@Override
-	public boolean isValid(CharSequence value, ConstraintValidatorContext constraintValidatorContext) {
-		if ( value == null ) {
-			return true;
-		}
-		Matcher m = pattern.matcher( value );
-		return m.matches();
```

Comment:
Maybe you could even keep this method here (well a generified version of it) and have a `CharSequence getCharSequenceRepresentation(T value)` method to override?